demos/deblurring_demo/python/model.py

```diff
@@ -0,0 +1,107 @@
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from pathlib import Path
+import sys
+
+import math
+import numpy as np
+
+sys.path.append(str(Path(__file__).resolve().parents[2] / 'common/python'))
+
+from models import Model
+
+class DeblurringModel(Model):
+    def __init__(self, ie, model_path, input_image_shape):
+        super().__init__(ie, model_path)
+        self.calculate_new_shape(input_image_shape)
+        self.input_blob_name = self.prepare_inputs()
+        self.output_blob_name = self.prepare_outputs()
+
+    def calculate_new_shape(self, shape):
+        h, w, _ = shape
+        block_size = 32
+        new_height = math.ceil(h / block_size) * block_size
+        new_width = math.ceil(w / block_size) * block_size
+
+        input_layer = next(iter(self.net.input_info))
+        input_shape = self.net.input_info[input_layer].input_data.shape
+        input_shape[2:] = (new_height, new_width)
+        self.net.reshape({input_layer: input_shape})
+
+    def prepare_inputs(self):
+        input_num = len(self.net.input_info)
+        if input_num != 1:
+            raise RuntimeError("Demo supports topologies only with 1 input")
+
+        input_blob_name = next(iter(self.net.input_info))
+        input_blob = self.net.input_info[input_blob_name]
+        input_blob.precision = "FP32"
+
+        input_size = input_blob.input_data.shape
+        if len(input_size) == 4 and input_size[1] == 3:
+            self.n, self.c, self.h, self.w = input_size
+        else:
+            raise RuntimeError("3-channel 4-dimensional model's input is expected")
+
+        return input_blob_name
+
+    def prepare_outputs(self):
+        output_num = len(self.net.outputs)
+        if output_num != 1:
+            raise RuntimeError("Demo supports topologies only with 1 output")
+
+        output_blob_name = next(iter(self.net.outputs))
+        output_blob = self.net.outputs[output_blob_name]
+        output_blob.precision = "FP32"
+
+        output_size = output_blob.shape
+        if len(output_size) == 4:
+            self.out_channels = output_size[1]
+            self.out_height = output_size[2]
+            self.out_width = output_size[3]
+        else:
+            raise Exception("Unexpected output blob shape {}. Only 4D output blob is supported".format(output_size))
+
+        return output_blob_name
+
+    def preprocess(self, inputs):
+        image = inputs
+        # right bottom padding to resize input image to input_layer shape
+        pad_params = {'mode': 'constant',
+                      'constant_values': 0,
+                      'pad_width': ((0, self.h - image.shape[0]), (0, self.w - image.shape[1]), (0, 0))
```

Comment:
Second image can be of different size compared to the first one. This may result in `pad_width` having negative values. Raise an exception with meaningful error or use cv2.resize() to `self.h, self.w`.

Reply:
I believe it's still aplicable.

Reply:
Fixed